src/hwmon.cpp

```diff
@@ -108,15 +149,7 @@ vector<string> HwmonInterface<HwmonT>::find_hwmons_by_name(
 		return result;  // don't recurse to subdirs
 	}
 
-	struct dirent **entries;
-	int nentries = ::scandir(path.c_str(), &entries, filter_subdirs, nullptr);
-	if (nentries == -1) {
-		return result;
-	}
-	for (int i = 0; i < nentries; i++) {
-		auto subdir = path + "/" + entries[i]->d_name;
-		free(entries[i]);
-
+	for (const filesystem::path &subdir : dir_entries<filter_subdirs>(path)) {
```

Comment:
```suggestion
	for (const auto &subdir : dir_entries<filter_subdirs>(path)) {
```

and elsewhere in this PR